packages/blocto-sdk/src/providers/ethereum.ts

```diff
@@ -839,27 +901,29 @@ export default class EthereumProvider
   ): Promise<string> {
     this.#checkNetworkMatched();
 
-    const extractParams = (params: Array<any>): Array<any> =>
-      params.map((param) =>
-        'params' in param
-          ? param.params[0] // handle passing web3.eth.sendTransaction.request(...) as a parameter with params
-          : param
-      );
-    const formatParams = extractParams(payload.params as Array<any>);
-    const copyPayload = { ...payload, params: formatParams };
+    // compatible web3.js old version sdk
+    const formatParams = Array.isArray(payload.params)
+      ? payload.params[0]
+      : payload.params;
```

Comment:
`payload.params` 應該永遠都是長 `[[{}], revert]`？因為我們在 L260 處理了 web3 v1 的資料
所以應該是要對 `payload.params[0]` 跑 `extractParams` 嗎？

Reply:
這個的意思是指 const [originalParams, revertFlag = false] = payload.params 因為我們幫 web3 v1 重包過了
然後原本的處理應該會變這樣
```js
    const extractParams = (params: Array<any>): Array<any> =>
      params.map((param) =>
        'params' in param
          ? param.params[0] // handle passing web3.eth.sendTransaction.request(...) as a parameter with params
          : param
      );
    const formatParams = extractParams(originalParams); // handle the params because web3 v1 might still send with web3.eth.sendTransaction.request(...) as the parameter
```